### PR TITLE
8289255: updated eclipse .classpath and other configuration files

### PR DESCRIPTION
**Pull request by andy-goryachev-oracle**
summary of changes:
- fixed .classpath files
- set UTF-8 encoding in the eclipse project config files
- fixed gradlew to use UTF-8 encoding

### CODE REVIEW

**Comment by kevinrushforth** [comment]
Once the OCA status has been verified, the review can proceed formally.

@nlisker Would you be able to review these changes? I'm not an Eclipse user.

@kleopatra your input would be helpful, too, if you are willing to take a look.

File: gradlew [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 #
-# Copyright � 2015-2021 the original authors.
+# Copyright © 2015-2021 the original authors.
```
Comment (kevinrushforth) [inline gradlew:4]: The changes in `gradlew` look fine. I'm surprised the file wasn't already UTF-8.

Comment (kevinrushforth) [reply gradlew:4]: Since this is a 3rd-party file, it's probably best to fix it separately the next time we update it, though.

Comment (andy-goryachev-oracle) [reply gradlew:4]: good point.

**Comment by nlisker** [comment]
There is already an open issue for this in [JDK-8221708](https://bugs.openjdk.org/browse/JDK-8221708). The reason it's pending is because the OpenJFX repo is awaiting [cleanup](https://mail.openjdk.org/pipermail/openjfx-dev/2021-June/030894.html), which will require to redo the Eclipse configuration files. The changes I have pending locally are more comprehensive than those proposed here. We can review these, but it will need to be redone later anyway.

**Comment by andy-goryachev-oracle** [comment]
@nlisker:
The goal of this PR is to unblock me for the further fixes in Eclipse, as the project currently does not compile in Eclipse.  I am sure there is more work in /apps, but at least it compiles now.

**Comment by nlisker** [comment]
By the way, I already filed a PR to fix these in the previous repo - https://github.com/openjdk/jfx/pull/514. No one wanted to review it so it just stayed there until the repo was defunct. That PR is probably not completely relevant now due to the partial cleanup.

**Comment by andy-goryachev-oracle** [comment]
Very similar indeed.  The latest version  2022-06 (4.24.0) complained more, so this PR touched more files.  I wonder if #514 will have zero diffs (save for XML ordering) once this PR is merged (integrated?).
I would definitely volunteer to review the remaining changes.

**Comment by nlisker** [comment]
> I would definitely volunteer to review the remaining changes (once I accumulate enough karma).

Maybe that's the trick. If I make the changes, it requires another Reviewer to allow me to merge, and there were (are?) no volunteers. If someone else makes the changes, I can count as a Reviewer :)

I am on 4.23. I will update to 4.24 and review it.

File: gradlew (hunk 1 above)
Comment (kevinrushforth) [reply gradlew:4]: So given that this is unrelated, please revert this file.

Comment (andy-goryachev-oracle) [reply gradlew]: reverted.  we do want to fix it, right?

File: gradlew [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 #
-# Copyright � 2015-2021 the original authors.
+# Copyright � 2015-2021 the original authors.
```
Comment (kevinrushforth) [inline gradlew:4]: `gradlew` still shows up in the patch with changes. You might need to revert it (e.g., using `git checkout master -- gradlew`) from the command line.

Comment (andy-goryachev-oracle) [reply gradlew]: done.

File: gradlew (hunk 1 above)
Comment (kevinrushforth) [reply gradlew]: Yes, we can do it the next time we update gradle (which will be in the JavaFX 20 time frame).

**Comment by nlisker** [comment]
I think that the latest commit is related to a different branch.

**Comment by andy-goryachev-oracle** [comment]
> I think that the latest commit is related to a different branch.

not sure why the last commit got pushed to this branch, will revert shortly.  thanks!

**Comment by nlisker** [comment]
I scheduled to review this today or tomorrow.

**Comment by andy-goryachev-oracle** [comment]
> I scheduled to review this today or tomorrow.

thank you

**Review by nlisker — CHANGES_REQUESTED** [review]
I found my branch that was dealing with the Eclipse files update: https://github.com/openjdk/jfx/compare/master...nlisker:jfx:8221708_Update_Eclipse_project_files

It contains changes to the apps project too. I compared your changes with mine, although mine are outdated in some places.

* You can remove the eclipse project and classpath files from `jfx/buildSrc`. It's a groovy project that is not very relevant. it can be manually imported if someone needs it.
* I'm still getting errors in tests/system with this PR:
```
Description	Resource	Path	Location	Type
AfterAll cannot be resolved to a type	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 78	Java Problem
Assertions cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 68	Java Problem
Assertions cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 159	Java Problem
Assertions cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 162	Java Problem
Assumptions cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 64	Java Problem
BeforeAll cannot be resolved to a type	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 72	Java Problem
ClipboardTransferable cannot be resolved to a type	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 132	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 51	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 57	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 57	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 75	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 76	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 91	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 92	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 102	Java Problem
MacPasteboardShim cannot be resolved to a type	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 103	Java Problem
SunClipboard cannot be resolved to a type	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 146	Java Problem
Test cannot be resolved to a type	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 62	Java Problem
The import com.sun.glass.ui.mac cannot be resolved	MacPasteboardTest.java	/tests/system/src/test/java/test/com/sun/glass/ui/mac	line 30	Java Problem
The import org.junit.jupiter cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 33	Java Problem
The import org.junit.jupiter cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 34	Java Problem
The import org.junit.jupiter cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 35	Java Problem
The import org.junit.jupiter cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 36	Java Problem
The import org.junit.jupiter cannot be resolved	DoubleShortcutProcessingTest.java	/tests/system/src/test/java/test/robot/javafx/scene	line 37	Java Problem
The import sun.awt.datatransfer cannot be resolved	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 44	Java Problem
The import sun.awt.datatransfer cannot be resolved	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 45	Java Problem
The method clearNativeContext() of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 192	Java Problem
The method getClipboardData(long) of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 207	Java Problem
The method getClipboardFormats() of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 202	Java Problem
The method getID() of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 187	Java Problem
The method registerClipboardViewerChecked() of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 216	Java Problem
The method setContentsNative(Transferable) of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 197	Java Problem
The method unregisterClipboardViewerChecked() of type ClipboardTest.CustomClipboard must override or implement a supertype method	ClipboardTest.java	/tests/system/src/test/java/test/javafx/scene/input	line 221	Java Problem
The type com.sun.webkit.WebPage is not accessible	PageFillTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 28	Java Problem
The type com.sun.webkit.WebPage is not accessible	StraightLineTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 28	Java Problem
The type com.sun.webkit.WebPage is not accessible	WebPageTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 41	Java Problem
The type com.sun.webkit.WebPageShim is not accessible	PageFillTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 29	Java Problem
The type com.sun.webkit.WebPageShim is not accessible	StraightLineTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 29	Java Problem
The type com.sun.webkit.WebPageShim is not accessible	WebPageTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 42	Java Problem
WebPage cannot be resolved to a type	PageFillTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 184	Java Problem
WebPage cannot be resolved to a type	PageFillTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 237	Java Problem
WebPage cannot be resolved to a type	WebPageTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 138	Java Problem
WebPageShim cannot be resolved	PageFillTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 186	Java Problem
WebPageShim cannot be resolved	PageFillTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 239	Java Problem
WebPageShim cannot be resolved	WebPageTest.java	/tests/system/src/test/java/test/javafx/scene/web	line 140	Java Problem
```
Do you not get these?

File: modules/javafx.graphics/.classpath [hunk 3]
```diff
@@ -1,17 +1,7 @@
...
-			<attribute name="ignore_optional_problems" value="true"/>
-		</attributes>
-	</classpathentry>
 	<classpathentry kind="src" path="build/gensrc/jsl-prism"/>
-	<classpathentry kind="src" path="build/hlsl/Decora">
-		<attributes>
-			<attribute name="ignore_optional_problems" value="true"/>
-		</attributes>
-	</classpathentry>
 	<classpathentry kind="src" path="build/gensrc/jsl-decora"/>
```
[the remark this answers is not in the document]
Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/.classpath:5]: 1. I will try to incorporate your changes, especially in apps/.  My initial goal was to fix the build of everything except apps/
2. I do get some warnings, they will be addressed by other fixes under JDK-8289379.  But others (ex: MacPasteboardShim) I do not see, as those have been fixed.  Could you please make sure you have no modified files and do a refresh?
3. Could you clarify your point about build/hlsl/Decora and build/hlsl/Prism please?  I noticed there is nothing there, and nothing seem to be put there.  Also, there are build/gensrc/jsl-prism and build/gensrc/jsl-decora directories being created, so I felt we can remove these entries from .classpath.  I do build on Mac though.

File: modules/javafx.swing/.classpath [hunk 4]
```diff
@@ -1,6 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="src" output="testbin" path="src/shims/java">
+		<attributes>
+			<attribute name="test" value="true"/>
+		</attributes>
+	</classpathentry>
 	<classpathentry kind="src" output="testbin" path="src/test/java">
```
[the remark this answers is not in the document]
Comment (andy-goryachev-oracle) [reply modules/javafx.swing/.classpath:9]: line 4 is <classpathentry kind="src" output="testbin" path="src/shims/java">

shims seem to be required, notice you get a bunch of shim-related warnings above.

[the remark this answers is not in the document]
Comment (nlisker) [reply modules/javafx.swing/.classpath:9]: I marked line 9, which is `src/test/java`. This folder does not exist in the repo, but the classpath file defines it. It creates an error. Do you not get this error?

File: modules/javafx.graphics/.classpath (hunk 3 above)
[the remark this answers is not in the document]
Comment (nlisker) [reply modules/javafx.graphics/.classpath:5]: 1. The way the apps are structured is that each app is its own project, which means that there are many new projects to create and configure. It can be a bit of work, though I have done it in my branch already. If you want to leave it to round 2 of fixing the Eclipse files it's fine.

2+3. I worked directly from master and from your branch, so no files have been changed. The folders under `build` are OS-specific and are generated as needed by the build. Since you are on Mac, it would explain why the hlsl folders are empty, but they are not empty on Windows. This is why they are marked as "ignore_optional_problems" - to not produce errors even on the wrong system. Since they don't contain Java classes that are referenced by the build (as far as I could tell), it should be safe to remove the `build/...` source folders in general. I don't know if on Mac you get any issues, but I assume you don't.

**Comment by andy-goryachev-oracle** [comment]
Note: tests are failing due to JDK-8289751

File: modules/javafx.graphics/.classpath (hunk 3 above)
[the remark this answers is not in the document]
Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/.classpath:5]: 1. I would rather leave the apps/ for day 2, if I may.
2+3.  Currently, on Mac, I don't see any issues related to build/... folders.  I think, generally speaking, it's a bad idea to refer to derived folders as source - it complicates the build process in Eclipse.  Does my branch fail on windows in Eclipse, Nir?  I mean, do I need to make changes to address this comment?

File: modules/javafx.swing/.classpath (hunk 4 above)
[the remark this answers is not in the document]
Comment (nlisker) [reply modules/javafx.swing/.classpath]: > there is nothing in javafx.swing/src/test/java (no tests??) git does not version directories, unlike some other VCs.

I think I understand now why you didn't see the error. You had an empty folder, for some reason, that made Eclipse satisfied with the classpath file entry. The folder does not exist in the repo, so I expect anyone who clones it to get an error for a missing folder. This is why I removed it in my branch.

File: modules/javafx.graphics/.classpath (hunk 3 above)
[the remark this answers is not in the document]
Comment (nlisker) [reply modules/javafx.graphics/.classpath:5]: I don't get any errors on the graphics module, I'm suggesting that you remove the entries:
```
	<classpathentry kind="src" path="build/gensrc/jsl-prism"/>
	<classpathentry kind="src" path="build/gensrc/jsl-decora"/>
```

[the remark this answers is not in the document]
Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/.classpath:5]: good point.  removing

	<classpathentry kind="src" path="build/gensrc/jsl-prism"/>
	<classpathentry kind="src" path="build/gensrc/jsl-decora"/>

**Comment by andy-goryachev-oracle** [comment]
@nlisker are the changes acceptable or do we need more work?  I'd like to merge this PR because otherwise I have to cherry pick to submit other PRs, and cherry picking is not a good idea, generally speaking.
thank you.

**Comment by nlisker** [comment]
You didn't address my [review comment](https://github.com/openjdk/jfx/pull/804#pullrequestreview-1026557434) yet.

**Comment by nlisker** [comment]
In addition, now I'm getting an error on `apps/samples/3DViewer`: `The type java.lang.Class cannot be resolved. It is indirectly referenced from required .class files`

**Comment by nlisker** [comment]
> This PR does not fix apps/ project. In fact, I think .project in apps/ should be removed, and instead we should create individual projects, either for a group (i.e. samples, toys), or for each application (i.e. Robot, 3DViewer).
> 
> This work depends on the cleanup. We could probably do it in a separate bug. What do you think?

That's fine, but I saw that you changed the 3DViewer and Ensemble8 files, and there are still errors there, so either leave apps completely out of this PR or bring it to a point where there are no errors.

As in my branch that I linked above in the review comment, I also created a separate project for each app. The `apps` project itself still needs a .project file because it's a gradle project that is created automatically by the gradle build. It doesn't need to be a Java project though, so the .project file can have the gradle nature only.

**Comment by andy-goryachev-oracle** [comment]
I did have samples project open - that would explain my changes in 3DViewer and Ensemble8 files.
Let me see what it would take to create individual projects for toys and such.
I will change the nature of the apps .project to gradle.

**Comment by nlisker** [comment]
> Let me see what it would take to create individual projects for toys and such.

You can look at my branch, it does exactly that, and I added all the module reads/exports (unless something changed).

**Comment by andy-goryachev-oracle** [comment]
Nir:
I've merged your branch into mine (ag.apps, not yet pushed).  
Two things:
1. Even though apps/.project has no nature, opening it in eclipse results in 30K errors (class not found)
2. when I close apps/ project, I get the following errors in systemTests:

Description	Location	Path	Resource
The package myapp3.pkg2 does not exist or is empty	line 33	/systemTests/src/testapp3/java/mymod	module-info.java
The package myapp3.pkg3 does not exist or is empty	line 34	/systemTests/src/testapp3/java/mymod	module-info.java
The package myapp3 does not exist or is empty	line 30	/systemTests/src/testapp3/java/mymod	module-info.java
The project was not built due to "module-info [in module-info.java [in mymod [in src/testapp2/java [in systemTests]]]] does not exist". Fix the problem, then try refreshing this project and building it since it may be inconsistent	Unknown		systemTests

Do we really need a .project file in apps/?

**Comment by nlisker** [comment]
Gradle creates apps automatically as a gradle project, so it needs the gradle nature. I can have a look later again. Do you use Buildship? Maybe that's a source of differences as I certainly don't get thousands of errors.

**Comment by andy-goryachev-oracle** [comment]
I am not using gradle to build in eclipse.  the expectation is that I can open apps project and it would at least compile.
but right now, even though the .project file in apps/ has no nature whatsoever, it seems to be compiled as java, or at least all java sources show errors.

Here is .project in apps/ :
<?xml version="1.0" encoding="UTF-8"?>
<projectDescription>
        <name>apps</name>
        <comment></comment>
        <projects>
        </projects>
        <buildSpec>
        </buildSpec>
        <natures>
        </natures>
</projectDescription>

**Comment by andy-goryachev-oracle** [comment]
github ate the file.  here is the .project in apps/ after I merged your branch:

`<?xml version="1.0" encoding="UTF-8"?>
<projectDescription>
        <name>apps</name>
        <comment></comment>
        <projects>
        </projects>
        <buildSpec>
        </buildSpec>
        <natures>
        </natures>
</projectDescription>`

**Comment by nlisker** [comment]
Ah, that explains it. I use Buildship which adds the gradle nature. I don't know how eclipse handles projects with no nature, I will have to look at that. It also means that there are going to be conflicts between those who use Buildship and those who don't.

**Comment by andy-goryachev-oracle** [comment]
I have gradle and buildship installed.  Is there a step missing, after some one checks out the jfx repo?

**Comment by andy-goryachev-oracle** [comment]
No, my workspace (jfx) was imported as a gradle project.  I think it created a bunch of .project and .classpath files that *never worked* - I could not get them to compile.  I had to reset all of them, and now I have 0 errors and 0 warnings with the apps project closed.

At this point, I am not sure why you see these errors - do you have any files modified/created?  

Could you close the apps project in eclipse and do Project -> Clean followed by Project -> Build All?

**Comment by nlisker** [comment]
> No, my workspace (jfx) was imported as a gradle project. I think it created a bunch of .project and .classpath files that never worked - I could not get them to compile. I had to reset all of them

That's what I wrote in the instructions - gradle creates its own project files that are wrong (it's not smart enough to handle modular projects yet), and they need to be reverted to those in the repo, or in this PR in this case.

I updated to the latest commit and imported as Gradle projects in a new workspace (and reverted to this PR's Eclipse files). I still get an error on 3DViewer: `The type java.lang.Class cannot be resolved. It is indirectly referenced from required .class files`. This error points at some configuration issue. If you don't get this error for some reason then we can continue and get there when we do the apps folder, otherwise it should be fixed.

In addition, If I import as Java projects (not as Gradle), `buildSrc` still has project files that should be removed (and apps is a mess, which is fine for now).
`tests` also has project files that need to be fixed (`tests` includes the Gradle project `systemTests`) if they are imported this way. The errors are the ones I posted in my review comment. If you want to work on `tests` on a later PR that's also fine because there is some work to do there.

**Comment by andy-goryachev-oracle** [comment]
Nir, I think I understand what the problem is.

For the purposes of this PR, buildSrc, apps(toys, samples), tests are not included.  (The purpose of this PR is to be able to compile in Eclipse and get to 0 warnings for essential parts).

systemTests *is* included, as it contains system tests, and I am not sure how eclipse actually handles this kind of nested .projects's (tests/.project and tests/system/.project).  I will update this PR such that tests/ can be included instead of systemTests/.

apps/ will require more work, and is also impacted by the cleanup effort you and Kevin discussed in the related tickets.  I would certainly like to participate in this effort, under a separate ticket.

To summarize, the list of eclipse projects that do compile in Eclipse as a result of this PR are:
- base
- controls
- fxml
- graphics
- jfx
- media
- swing
- swt
- systemTests (can be dropped in favor ot tests
- tests
- web

Not supported (should be removed from workspace or simply closed):
- apps
- apps/*
- buildSrc

**Comment by nlisker** [comment]
Since apps are not included in the goal of this PR, please revert the changes made to them. I will review the changes to `tests` and then the PR can be integrated if there are no further issues. A followup issue should be filed for `apps` in JBS if there is none.

**Comment by nlisker** [comment]
@kevinrushforth 2 questions that are relevant to the future work on the Eclipse files:
1. The jfx gradle build treats `tests/system` as a project but not its parent `test`. The cleanup talked about wiring up `manual` and `performance` to the build. Do you have an idea when the work on this will start?
2. The cleanup also talked about moving `apps/toys` and `apps/tests` to `tests/manual`. This means that `apps/samples` is the only folder to remain in `apps` (`apps/performance` was removed already). Is there a point to keeping the hierarchy? Maybe just `apps` or just `samples` will be sufficient in this case.

**Comment by andy-goryachev-oracle** [comment]
> Since apps are not included in the goal of this PR, please revert the changes made to them.

reverted.

**Review by nlisker — APPROVED** [review]
It might be a good idea to add the JUnit 5 container to all modules preemptively so that when someone adds a test with Junit 5 we won't need to patch the classpath file ad-hoc. Fine to leave as is.

**Comment by andy-goryachev-oracle** [comment]
> It might be a good idea to add the JUnit 5 container to all modules preemptively so that when someone adds a test with Junit 5 we won't need to patch the classpath file ad-hoc. Fine to leave as is.

I agree.  Let's do it as a part of apps/
